bokeh/models/tools.py

```diff
@@ -413,3 +413,13 @@ class HoverTool(Tool):
         the ``Rect`` and ``Quad`` glyphs are supported.
 
     """)
+
+    mode = Enum("point", "hline", "vline", help="""
+    Whether to consider hover pointer as point (x/y values), or a
+    span on h or v directions.
+    """)
+
+    hit_value_mode = Enum("snap_to_data", "glyph_center", "hit_interpolate",
```

Comment:
yes.. it's a bit hard trying to "make it work" well for line and all other glyphs... There are 2 aspects: tooltip position and value (that could be "continuous"/interpolated or snap to data). If we agree on sticking those together for lines then we can probably cover all cases with your proposal to have:

```
snap_to_data = Bool()  # False is equiv to "follow_mouse"
allow_interpolation = Bool() # only has potential effect for lines 
```